src/symbol/projection.js

```diff
@@ -87,31 +89,61 @@ const maxTangent = Math.tan(85 * Math.PI / 180);
 
 /*
  * Returns a matrix for converting from tile units to the correct label coordinate space.
+ * This variation of the function returns a label space matrix specialized for rendering.
+ * It transforms coordinates as-is to whatever the target space is (either 2D or 3D).
+ * See also `getLabelPlaneMatrixForPlacement`
  */
-function getLabelPlaneMatrix(posMatrix: Float32Array,
+function getLabelPlaneMatrixForRendering(posMatrix: Float32Array,
                              tileID: CanonicalTileID,
                              pitchWithMap: boolean,
                              rotateWithMap: boolean,
                              transform: Transform,
                              pixelsToTileUnits: Float32Array): Float32Array {
     const m = mat4.create();
+
     if (pitchWithMap) {
         if (transform.projection.name === 'globe') {
-            mat4.multiply(m, m, calculateGlobeLabelMatrix(transform, tileID));
-
+            const lm = calculateGlobeLabelMatrix(transform, tileID);
+            mat4.multiply(m, m, lm);
         } else {
             const s = mat2.invert([], pixelsToTileUnits);
             m[0] = s[0];
             m[1] = s[1];
             m[4] = s[2];
             m[5] = s[3];
-        }
-        if (!rotateWithMap) {
-            mat4.rotateZ(m, m, transform.angle);
+            if (!rotateWithMap) {
+                mat4.rotateZ(m, m, transform.angle);
+            }
         }
     } else {
         mat4.multiply(m, transform.labelPlaneMatrix, posMatrix);
     }
+
+    return m;
+}
+
+/*
+ * Returns a matrix for converting from tile units to the correct label coordinate space.
+ * This variation of the function returns a matrix specialized for placement logic.
+ * Coordinates will be clamped to x&y 2D plane which is used with viewport and map aligned placement
+ * logic in most cases. Certain projections such as globe view will use 3D space for map aligned
+ * label placement.
```

Comment:
Nit> could be useful to comment why using the same matrix for e.g. placement of !pitchWithMap viewport aligned symbols would be incorrect. I mean, although failing render tests clarify this very well, a comment to a first time reader, explaining why we have two variants of the same matrix, one specialized or placement and another for rendering would help.

Reply:
> Do you think I should be more explicit about this in the comments?